src/LinAlg/hiopMatrix.hpp

```diff
@@ -84,6 +84,16 @@ class hiopMatrix
   virtual hiopMatrix* alloc_clone() const=0;
   virtual hiopMatrix* new_copy() const=0;
 
+  virtual void copyFrom(const hiopMatrix& matrix) = 0;
+  virtual void copyFrom(const double* buffer) = 0;
+
+  /* copies 'src' into this as a block starting at (i_block_start,j_block_start) */
+  void copyBlockFromMatrix(const long i_block_start, const long j_block_start, const hiopMatrix& src) = 0;
```

Comment:
this method makes sense only to particular matrices (dense and maybe mds), and I would not make it part of the interface.